solidity/contracts/bridge/Bridge.sol

```diff
@@ -264,34 +323,309 @@ contract Bridge {
+        value = output.extractValue();
+        bytes memory walletPubKeyHashBytes = output.extractHash();
+        // The sweep transaction output should always be P2PKH or P2WPKH.
+        // In both cases, the wallet public key hash should be 20 bytes length.
+        require(
+            walletPubKeyHashBytes.length == 20,
+            "Wallet public key hash should have 20 bytes"
+        );
+        /* solhint-disable-next-line no-inline-assembly */
+        assembly {
+            walletPubKeyHash := mload(add(walletPubKeyHashBytes, 32))
+        }
+
+        return (walletPubKeyHash, value);
+    }
+
+    // TODO: Documentation.
+    function processSweepTxInputs(
+        BitcoinTx.Info calldata sweepTx,
+        bytes20 walletPubKeyHash
+    )
+        internal
+        returns (
+            uint256 inputsValue,
+            address[] memory depositors,
+            uint256[] memory depositedAmounts
+        )
+    {
+        SweepInfo storage previousSweep = sweeps[walletPubKeyHash];
+        // Flag indicating whether the previous sweep transaction output
+        // is included as an input. If current sweep is the first sweep
+        // of given wallet, we use a shortcut and initialize it with `true`.
+        bool previousSweepFlag = previousSweep.txHash == bytes32(0);
+
+        // Determining the total number of sweep transaction inputs in the same
+        // way as for number of outputs.
+        (uint256 inputsCompactSizeUintLength, uint256 inputsCount) =
+            sweepTx.inputVector.parseVarInt();
+
+        // To determine the first input starting index, we must jump over
+        // the compactSize uint which prepends the input vector. One byte must
+        // be added because of how `parseVarInt` returns the length of the
+        // compactSize uint. Refer `BTCUtils` library for more details.
+        uint256 inputStartingIndex = 1 + inputsCompactSizeUintLength;
+
+        // Determine the swept deposits count. If the `previousSweepFlag` is
+        // already `true` that means the current sweep is the first one
+        // for given wallet and we shouldn't expect an additional input. If
+        // so, all inputs should be deposits. Otherwise, we need to subtract
+        // one input since it represents a previous sweep output.
+        uint256 depositsCount =
+            previousSweepFlag ? inputsCount : inputsCount - 1;
+
+        // Initialize helper variables.
+        uint256 processedDepositsCount = 0;
+        depositors = new address[](depositsCount);
+        depositedAmounts = new uint256[](depositsCount);
+
+        // Inputs processing loop.
+        for (uint256 i = 0; i < inputsCount; i++) {
+            // Check if we are at the end of the input vector.
+            if (inputStartingIndex >= sweepTx.inputVector.length) {
+                break;
+            }
+
+            (bytes memory input, uint256 inputLength) =
+                extractTxInput(sweepTx, inputStartingIndex);
```

Comment:
It should be possible to change this so that we don't need to allocate the `bytes memory input` at all. I'll make sure that `bitcoin-spv` has the functions we'll need.

Reply:
With https://github.com/keep-network/bitcoin-spv/pull/3/commits/d9248be6f752d636bd3ea12f1fe05aadee6c8cd5 we can use `extractInputTxIdLeAt`, `extractTxIndexLeAt` and `determineInputLengthAt` to replace the `extractTxInput` call altogether.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/106/commits/152766279380c868726c81283ee1f7e8f33751fc though I need to keep it within `parseTxInputAt` to reduce local variables count to avoid `stack too deep` errors.